components/governance/org.wso2.carbon.governance.taxonomy/src/main/java/org/wso2/carbon/governance/taxonomy/services/StorageProviderImpl.java

```diff
@@ -0,0 +1,173 @@
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.governance.taxonomy.services;
+
+import org.apache.commons.io.IOUtils;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.governance.taxonomy.beans.QueryBean;
+import org.wso2.carbon.governance.taxonomy.beans.TaxonomyBean;
+import org.wso2.carbon.governance.taxonomy.internal.ServiceHolder;
+import org.wso2.carbon.governance.taxonomy.util.CommonUtils;
+import org.wso2.carbon.registry.core.Collection;
+import org.wso2.carbon.registry.core.Registry;
+import org.wso2.carbon.registry.core.exceptions.RegistryException;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.xml.sax.SAXException;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.xml.parsers.ParserConfigurationException;
+import static org.wso2.carbon.registry.core.RegistryConstants.GOVERNANCE_COMPONENT_PATH;
+
+/**
+ * this class will implements methods to manage tenant specific taxonomy data map
+ */
+class StorageProviderImpl implements IStorageProvider {
+    private static Map<Integer, Map<String, TaxonomyBean>> tenantTaxonomyMap;
+    private int tenantId;
+
+    StorageProviderImpl() {
+        tenantId = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId();
+    }
+
+    /**
+     * This method will return taxonomy document bean object for a given taxonomy taxonomyName from the map
+     *
+     * @param taxonomyName String taxonomyName of the taxonomy (this must be unique for a tenant)
+     * @return Taxonomy document bean which contains taxonomy meta data
+     */
+    @Override
+    public TaxonomyBean getTaxonomy(String taxonomyName) {
+        if (tenantTaxonomyMap != null) {
+            if (tenantTaxonomyMap.containsKey(tenantId)) {
+                return tenantTaxonomyMap.get(tenantId).get(taxonomyName);
+            }
+        }
+        return null;
+    }
+
+    /**
+     * This method will return taxonomy document bean object for a given taxonomy taxonomyName from the map
+     *
+     * @param taxonomyQueryBean taxonomy query meta data
+     * @return Taxonomy document bean which contains taxonomy meta data
+     */
+    @Override
+    public TaxonomyBean getTaxonomy(QueryBean taxonomyQueryBean) {
+        if (tenantTaxonomyMap != null) {
+            if (tenantTaxonomyMap.containsKey(tenantId)) {
+                return tenantTaxonomyMap.get(tenantId).get(taxonomyQueryBean.getTaxonomyName());
+            }
+        }
+        return null;
+    }
+
+    /**
+     * This method will remove taxonomy document bean object from the map for a given taxonomy name
+     *
+     * @param taxonomyName String taxonomy name
+     */
+    @Override
+    public void removeTaxonomy(String taxonomyName) {
+        if (tenantTaxonomyMap != null) {
+            if (tenantTaxonomyMap.containsKey(tenantId)) {
+                tenantTaxonomyMap.get(tenantId).remove(taxonomyName);
```

Comment:
Thank you Thushara for the advice.
Two line combined into one line. And did't have a separate method since its not using any other place. Fixed with this PR https://github.com/wso2/carbon-governance/pull/255